packages/features/ee/workflows/pages/index.tsx

```diff
@@ -37,20 +43,59 @@ function WorkflowsPage() {
     },
   });
 
+  const query = trpc.viewer.workflows.getByViewer.useQuery();
+
+  useEffect(() => {
+    const allWorkflows = allWorkflowsData?.workflows;
+    if (allWorkflows && allWorkflows.length > 0) {
+      const filtered = allWorkflows.filter((workflow) => {
+        if (!!workflow.userId && workflow.userId === checkedFilterItems.userId) return workflow;
+        if (checkedFilterItems.teamIds.includes(workflow.teamId || 0)) return workflow;
+      });
+      setFilteredWorkflows(filtered);
+    } else {
+      setFilteredWorkflows([]);
+    }
+  }, [checkedFilterItems, allWorkflowsData]);
+
+  useEffect(() => {
+    if (session.status !== "loading" && !query.isLoading) {
+      if (!query.data) return;
+      setCheckedFilterItems({
+        userId: session.data?.user.id || null,
+        teamIds: query.data.profiles
+          .map((profile) => {
+            if (!!profile.teamId) {
+              return profile.teamId;
+            }
+          })
+          .filter((teamId) => !!teamId) as number[],
```

Comment:
```suggestion
          .filter((teamId): teamId is number => !!teamId),
```

See https://medium.com/devil-is-in-the-details/filtering-arrays-using-type-predicates-in-typescript-cf554afc76a3

Reply:
Also. I think filtering profiles with teamId first and than capturing their teamId makes more sense.

Reply:
> Also. I think filtering profiles with teamId first and than capturing their teamId makes more sense.

@hariombalhara I am struggling to understand what you mean by that 🤔 could you maybe elaborate on that?

Reply:
Sorry for not being clear.
I was suggesting doing this. It filters out profiles with teamId first and then we extract teamId.
```javascript
query.data.profiles.filter((profile)=>!!profile.teamId).map((profile)=>profile.teamId)
```

Right now, we map all profiles and profiles with no `teamId` get mapped to `undefined` and then we filter out those that are not undefined. 

It's absolutely a nit. So, you can ignore it if you want.